### PR TITLE
[NU-1506] refactoring: ParameterName data class instead of String

### PR DESCRIPTION
**Pull request by mk-software-pl**
## Describe your changes

`ParameterName` data class instead of `String` as a representation of a parameter name

## Checklist before merge
- [x] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/parameters/SchemaBasedParameter.scala [hunk 1]
```diff
@@ -28,7 +29,7 @@ sealed trait SchemaBasedParameter {
 
 object SchemaBasedParameter {
   type RecordFieldName = String
-  type ParameterName   = String
+  type ParameterName   = pl.touk.nussknacker.engine.api.parameter.ParameterName
```
Comment (bartektartanus) [inline utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/parameters/SchemaBasedParameter.scala:32]:
```suggestion
  type ParameterName   = ParameterName
```

**Review by arkadius — APPROVED** [review]
LGTM

File: utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/parameters/SchemaBasedParameter.scala (hunk 1 above)
Comment (mk-software-pl) [reply utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/parameters/SchemaBasedParameter.scala:32]: I removed the type def. It's not needed now.

File: designer/client/cypress/e2e/__image_snapshots__/electron/Linux/Process should zoomrestore node window with test data #0.png
[the remark this answers is not in the document]
Comment (mk-software-pl) [reply designer/client/cypress/e2e/__image_snapshots__/electron/Linux/Process should zoomrestore node window with test data #0.png:1]: eh, right - it doesn't seem it's ok. I reverted it but for some reason, it's generated.

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/ParameterNaming.scala [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 
-  def getNameForBranchParameter(parameter: Parameter, branchId: String): String = {
+  def getNameForBranchParameter(parameter: Parameter, branchId: String): ParameterName = {
     getNameForBranchParameter(parameter.name, branchId)
   }
 
-  def getNameForBranchParameter(parameterName: String, branchId: String): String = {
-    s"$parameterName for branch $branchId"
+  def getNameForBranchParameter(parameterName: ParameterName, branchId: String): ParameterName = {
+    ParameterName(s"${parameterName.value} for branch $branchId")
```
[the remark this answers is not in the document]
Comment (mk-software-pl) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/ParameterNaming.scala:13]: Are you talking about adding an optional `branchId` field to the `ParameterName` data class?

[the remark this answers is not in the document]
Comment (bartektartanus) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/ParameterNaming.scala:13]: Yes, I was thinking about something like this:
```
final case class ParameterName(value: String, branchId: Option[String]) {
  def name(): branchId.map(b => s"$value for branch $b").getOrElse(value)
}
```

but leaving class definition as-is and adding object:
```
object ParameterName {
  def apply(value: String) = ParameterName(value)
  def withBranchId(value: String, branchId: String) = ParameterName(s"${parameterName.value} for branch $branchId")
} 
```
is also fine

**Review by bartektartanus — APPROVED** [review]

**Pull request merged**